core/impl/src/main/java/pt/webdetails/cdf/dd/model/meta/reader/cdexml/fs/XmlFsPluginModelReader.java

```diff
@@ -142,16 +143,16 @@ private void readPropertiesFile( MetaModel.Builder model, XmlFsPluginThingReader
 
     // One file can contain multiple definitions.
     @SuppressWarnings( "unchecked" )
-    List<Element> propertieElems = doc.selectNodes( "//DesignerProperty" );
-    for ( Element propertyElem : propertieElems ) {
-      readProperty( model, factory, propertyElem, file );
+    List<Node> nodes = doc.selectNodes( "//DesignerProperty" );
```

Comment:
You should downcast to List<Element> in this case, to avoid changing the signature of readProperty and corresponding cast to `DefaultElement` below... (see how in: https://stackoverflow.com/a/15427707/178749).

Also, why not use `org.dom4j.Element` and instead `org.dom4j.tree.DefaultElement`?